node/resources/addProductTranslationsCSV.ts

```diff
@@ -1,8 +1,108 @@
+import { SaveArgs as TranslateMessageArgs } from '@vtex/api'
 import parse from './utils/parser'
 
-async function addProductTranslationsCSV(csv: FileUpload, _: Context): Promise<void> {
-  console.log(`------> await parse()`, JSON.stringify([await parse(csv)], null, 2))
-  return 
+type MessagesOfProvider = TranslateMessageArgs['messagesByProvider'][0]
+
+enum TranslatableField {
+  description = 'description',
+  name = 'name',
+}
+
+const ID_CSV_DESC = '_ProductId'
+
+const FIELDS_TO_CSV_DESC : Record<TranslatableField, string> = {
+  'description': '_ProductDescription',
+  'name': '_ProductName',
+}
+
+async function addProductTranslationsCSV(csv: FileUpload, translateTo: string, ctx: Context): Promise<string[]> {
+  const { messagesGraphQL } = ctx.clients
+
+  const [headers, ...data] = await parse(csv)
+  const fieldToIndex: Partial<Record<TranslatableField, number>> = {}
+
+  const keys = Object.keys(TranslatableField) as TranslatableField[]
+  keys.forEach(key => {
+    const desc = FIELDS_TO_CSV_DESC[key]
+    const index = headers.findIndex(header => header.startsWith(desc))
+    if (index !== -1) {
+      fieldToIndex[key] = index
+    }
+  })
+
+  const idIndex = headers.findIndex(header => header.startsWith(ID_CSV_DESC))
+  const fieldAndIndex = Object.entries(fieldToIndex)
+
+  const messagesByProvider: MessagesOfProvider[] = data.map(
+    row => ({
+      messages: fieldAndIndex.map(([key, index]) => ({
+        content: row[index!],
+        description: '',
+        id: key,
+      })),
+      provider: `Product-id.${row[idIndex]}`,
+    })
+  )
+  
+  const results = await Promise.all(
+    messagesByProvider.map(provider =>
+      messagesGraphQL
+        .save({
+          messagesByProvider: [provider],
+          to: translateTo,
+        })
+        .then(success => [success, provider] as [boolean, MessagesOfProvider])
+    )
+  )
+
+  const failures = results.filter(([success]) => !success)
+  const getProductId = ({ provider }: MessagesOfProvider) => provider.split('.')[1]
+  return failures.map(([_, messages]) => getProductId(messages))
+}
+
+export async function mockAddProductTranslationsCSV(_: FileUpload, translateTo: string, ctx: Context): Promise<string[]> {
```

Comment:
what ?

Reply:
Just a mock to time how long it takes to translate a lot of messages, since I don't have access to real prod data.